core/Symbols.cc

```diff
@@ -848,6 +848,11 @@ string Symbol::show(const GlobalState &gs) const {
         }
     }
 
+    // Make sure that we get nice erorr messages for things involving the proc sig builders.
+    if (this->name == core::Names::Constants::DeclBuilderForProcs()) {
+        return "T.proc";
```

Comment:
Another option here:

```suggestion
        return "T::Private::Methods::DeclBuilder";
```

up to you

Reply:
This one only shows up for uses of `T.proc`, so I thought that using the name that caused the error would be helpful. My reasoning was that it's a little bit like `T.attached_class` this way.